### PR TITLE
feat: robust(er) adhoc query validation

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

The function `has_table_query` is used to detect if a given query is selecting from one or more tables, in order to prevent malicious ad-hoc expressions that bypass RLS. Because `sqlparse` is non-validating the function needs to keep track of a lot of state, and often fails on edge cases.

This PR rewrites the function to use `sqloxide`. The advantage is that the function is now more robust, supporting custom dialects and handling more edge cases. There are a couple disadvantages, though:

- `sqloxide` was an optional dependency, with this PR it becomes mandatory.
- The function `has_table_query` now requires a full SQL query, so for validating expressions they need to be wrapped in `f"SELECT {expression}"`.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

The unit tests pass mostly unmodified, the only exception is that `table` is usually a reserved keyword and had to be quoted in order to `sqloxide` to parse it (`sqlparse` is much more lenient). I added a new test to cover another edge case that was found could bypass RLS.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: requirements/testing.txt [hunk 1]
```diff
@@ -107,6 +107,8 @@ pydata-google-auth==1.7.0
     # via pandas-gbq
 pyfakefs==5.2.2
     # via -r requirements/testing.in
+pyhive[presto]==0.6.5
+    # via apache-superset
```
Comment (betodealmeida) [inline requirements/testing.txt:111]: Not sure why this is showing up here now.

File: superset/connectors/sqla/models.py [hunk 2]
```diff
@@ -517,27 +517,6 @@ def data(self) -> Dict[str, Any]:
...
-                expression,
-                database_id,
-                schema,
-            )
-            expression = sanitize_clause(expression)
-        except (QueryClauseValidationException, SupersetSecurityException) as ex:
-            raise QueryObjectValidationError(ex.message) from ex
-    return expression
-
-
```
Comment (betodealmeida) [inline superset/connectors/sqla/models.py:540]: This method was almost identical to the base class, the only difference is that is handles `SupersetSecurityException`. I updated the base class method and removed this implementation.

**Comment by betodealmeida** [comment]
@villebro any thoughts on this?

**Comment by villebro** [comment]
@betodealmeida can you rebase this PR so we can get CI running? I'll be happy to review after that. I think it would be great to get this merged, it's a very useful improvement 👍

**Comment by sbelondr** [comment]
Hello,
I'm interested for this pr and I add this comment to be notified when this pr is validated. Thank you in advance if it can be validated quickly.
Thanks

**Comment by betodealmeida** [comment]
Closing due to https://github.com/apache/superset/issues/26786.
